### PR TITLE
Added phrases customizability

### PR DESCRIPTION
**Pull request by MaxxMXSZ**
Made a new pr since can't be bothered to resolve the conflict. this closes #299

### CODE REVIEW

**Review by ltzmax — COMMENTED** [review]
Should we also add catchphrase for shiny? Since i've seen people edit this so could be good too.

**Comment by MaxxMXSZ** [comment]
> Should we also add catchphrase for shiny? Since i've seen people edit this so could be good too.

Indeed would be useful will get it done later tomorrow

File: ballsdex/packages/countryballs/components.py [hunk 1]
```diff
@@ -87,25 +86,29 @@ async def on_submit(self, interaction: discord.Interaction["BallsDexBot"]):
...
+                message,
                 allowed_mentions=discord.AllowedMentions(users=player.can_be_mentioned),
             )
             self.button.disabled = True
             await interaction.followup.edit_message(self.ball.message.id, view=self.button.view)
         else:
-            await interaction.followup.send(
-                f"{interaction.user.mention} Wrong name!",
+            message = f"{interaction.user.mention} {settings.wrong_name_phrase}"
+            await interaction.response.send_message(
```
Comment (ltzmax) [inline ballsdex/packages/countryballs/components.py:110]: You gotta add back `await interaction.followup.send`. its from #430.

File: ballsdex/packages/countryballs/components.py [hunk 2]
```diff
@@ -174,7 +177,9 @@ def __init__(self, ball: "CountryBall"):
 
     async def callback(self, interaction: discord.Interaction):
         if self.ball.catched:
-            await interaction.response.send_message("I was caught already!", ephemeral=True)
```
Comment (ltzmax) [inline ballsdex/packages/countryballs/components.py:177]: Same here, gotta add back, related to #430.

**Review by ltzmax — CHANGES_REQUESTED** [review]

File: ballsdex/settings.py [hunk 3]
```diff
@@ -278,11 +337,53 @@ def update_settings(path: "Path"):
...
+        content += """
+# Override the phrase 'I was caught already!'
+caught-already-phrase: I was caught already!
+"""
+
+    if add_shiny_phrase:
+        content += """
+# override the phrase "✨ ***It's a shiny {settings.collectible_name}!*** ✨"
+# NOTE THAT you MUST include {collectible_name} somewhere in the phrase
+shiny-phrase: ✨ ***It's a shiny {settings.collectible_name}!***
```
Comment (ltzmax) [inline ballsdex/settings.py:386]:
```suggestion
# override the phrase "✨ ***It's a shiny {collectible_name}!*** ✨"
# NOTE THAT you MUST include {collectible_name} somewhere in the phrase
shiny-phrase: ✨ ***It's a shiny {collectible_name}!***
```

File: ballsdex/settings.py [hunk 4]
```diff
@@ -180,6 +210,29 @@ def write_default_settings(path: "Path"):
...
+# "This is a **new {collectible_name}** that has been added to your completion!"
+# NOTE THAT you MUST include {collectible_name} somewhere in the phrase
+new-completion-phrase: This is a **new {collectible_name}** that has been added to your completion!
+
+# override the phrase "I was caught already!"
+caught-already-phrase: I was caught already!
+
+# override the phrase "✨ ***It's a shiny {settings.collectible_name}!*** ✨"
+# NOTE THAT you MUST include {collectible_name} somewhere in the phrase
+shiny-phrase: ✨ ***It's a shiny {settings.collectible_name}!***
```
Comment (ltzmax) [inline ballsdex/settings.py:234]:
```suggestion
# override the phrase "✨ ***It's a shiny {collectible_name}!*** ✨"
# NOTE THAT you MUST include {collectible_name} somewhere in the phrase
shiny-phrase: ✨ ***It's a shiny {collectible_name}!***
```

**Review by ltzmax — CHANGES_REQUESTED** [review]

**Review by ltzmax — APPROVED** [review]
Looks good now. and it was already reviewed by flare on your previous pr on most part so we're good here.

**Review by ltzmax — CHANGES_REQUESTED** [review]
Well good we didn't merge yet. 
When we mistype the collectible_name, it'd actually just send errors + when the text is empty, it'd also happen. Maybe have a default text instead of None? just in case or something, @laggron42 maybe have better solutions to this!
```py
bot-1              | 2024-09-30 21:47:59 ERROR    ballsdex.core.bot Error in slash command spawn
bot-1              | Traceback (most recent call last):
bot-1              |   File "/usr/local/lib/python3.12/site-packages/discord/app_commands/commands.py", line 857, in _do_call
bot-1              |     return await self._callback(self.binding, interaction, **params)  # type: ignore
bot-1              |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
bot-1              |   File "/code/ballsdex/packages/admin/cog.py", line 534, in spawn
bot-1              |     result = await ball.spawn(channel or interaction.channel)  # type: ignore
bot-1              |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
bot-1              |   File "/code/ballsdex/packages/countryballs/countryball.py", line 60, in spawn
bot-1              |     f"{settings.wild_phrase.format(collectible_name=settings.collectible_name)}",
bot-1              |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
bot-1              | AttributeError: 'NoneType' object has no attribute 'format'
```

**Comment by MaxxMXSZ** [comment]
laggron seems to have a better solution i will leave this to him

**Pull request closed without merging**